File: go/longest-common-prefix.go
func longestCommonPrefix(strs []string) string {
	res := ""
	if len(strs) == 0 {
		return res
	}
	for i := 0; ; i++ {
		if i >= len(strs[0]) {
			return res
		}
		c := strs[0][i]
		for j := 1; j < len(strs); j++ {
			if i >= len(strs[j]) || strs[j][i] != c {
				return res
			}
		}
		res += string(c)
	}
	return res
}

File: cpp/longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string res = "";
        if (strs.size() == 0) return res;
        for (int i = 0; ; i++) {
            if (i >= strs[0].size()) return res;
            char c = strs[0][i];
            for (int j = 1; j < strs.size(); j++) {
                if (i >= strs[j].size() || strs[j][i] != c) return res;
            }
            res = res + c;
        }
        return res;
    }
};